app/src/main/java/com/gulderbone/cookieclicker/activities/ItemShop.kt

```diff
@@ -17,7 +17,7 @@ class ItemShop : AppCompatActivity() {
 
         upgradeButton = findViewById(R.id.upgrade)
 
-        upgradeButton.setOnClickListener { Game.producerPurchased(Grandma()) }
+        upgradeButton.setOnClickListener { Game.producerPurchased(Grandma(), this) }
```

Comment:
After discussing my codebase now I know that I should not send the entire context to functions.